_posts/2021-01-09-Julia-1.6-what-has-changed-since-1.0.md

```diff
@@ -0,0 +1,639 @@
+Full details can be found in the [NEWS.md](https://github.com/JuliaLang/julia/blob/backports-release-1.6/NEWS.md), and [HISTORY.md](https://github.com/JuliaLang/julia/blob/backports-release-1.6/HISTORY.md)
+
+
+## Front-end changes
+### Soft-scope in the REPL
+Julia 1.0 removed the notion of soft-scope from the language.
+I was very blasé about the change to for-loop bindings in my [1.0 release post](https://www.oxinabox.net/2018/06/01/Julia-Favourite-New-Things.html#for-loop-variable-binding-changes).
+Infact, I didn't even mention this particular change.
+It was [#19324](https://github.com/JuliaLang/julia/pull/19324) for reference
+
+This was undone in in Julia 1.5 with [#33864](https://github.com/JuliaLang/julia/pull/33864) for in the REPL only.
+Now in the REPL, assigning to a global variable within a for-loop actually assigns that variable, rather than shadowing it with a new variable in that scope.
+The same behavior outside the REPL, now gives a warning.
+
+Personally, this change never affected me because I never write for-loops at global scope that assign variables.
+Indeed basically all I code I write is inside functions.
+But I do see how this causes problems for some interactive work-flows, especially e.g. when demonstrating something.
+See the [main github issue](https://github.com/JuliaLang/julia/issues/28789) and the longest [Discourse thread](https://discourse.julialang.org/t/another-possible-solution-to-the-global-scope-debacle/15894), though there were many others.
+It took years of thinking to workout the solution, particularly because many of the more obvious solutions would be breaking in significant ways.
+
+### Deprecations are muted by default.
+
+This was me in [this PR](https://github.com/JuliaLang/julia/pull/35362).
+It's not something I am super-happy about.
+Though I think it makes sense.
+Using deprecated methods is actually fine, as long as your dependencies follow SemVer, and you use `^` (i.e. default) bounding in your Project.toml's `[compat]`.
+Which everyone does, because its enforced by the auto-merge script in the General registry.
+
+Solving deprecations is a thing you should chose to do, actively.
+Not casually, when trying to do something else.
+In particular, when updating to support a new major release of one of your dependencies, you should run your (integration) tests to see if they succeed.
+If they don't succeed, then one removed support for the new version and run julia with deprecation warnings turned on (or set to error); to identify and fix them.
+Probably also you are checking the release notes, since you are intentionally updating.
+
+The core of the reason is because it was actually breaking things.
+Irrelevant deprecation warning spam from dependencies of dependencies was causing JuMP and LightGraphs to become too slow to be used.
+Since they were from dependencies of dependencies the maintainers of those JuMP and LightGraphs couldn’t even fix them.
+Let alone the end users.
```

Comment:
```suggestion
Since they were from dependencies of dependencies the maintainers of JuMP and LightGraphs couldn’t even fix them, let alone the end users.
```